### PR TITLE
Add C implementation for `@stdlib/math/base/special/kernel-cos`

### LINKED ISSUE #645: [RFC]: Add C implementation for `@stdlib/math/base/special/kernel-cos`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/kernel-cos`](https://github.com/stdlib-js/stdlib/tree/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/kernel-cos).

```c
double stdlib_base_kernel_cos( const double x, const double y );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #645  .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All test and benchmarks passed, this PR is ready to be reviewed :)

**Comment by kgryte** [comment]
@Pranavchiku Should we reopen this? I believe this may have been auto-closed upon deleting the origin src rep.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md [hunk 1]
```diff
@@ -96,6 +96,99 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/special/kernel_cos.h"
+```
+
+#### stdlib_base_kernel_cos( x, y )
+
+Computes the [cosine][cosine] of a `number` on `[-π/4, π/4]`. For increased accuracy, the number for which the [cosine][cosine] should be evaluated can be supplied as a [double-double number][double-double-arithmetic] (i.e., a non-evaluated sum of two [double-precision floating-point numbers][ieee754] `x` and `y`).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md:127]: Move to note.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md [hunk 2]
```diff
@@ -96,6 +96,99 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+```c
+var v = stdlib_base_kernel_cos( 0.0, 0.0 );
+// returns ~0.0
+v = stdlib_base_kernel_cos( 3.141592653589793/6.0, 0.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md:132]:
```suggestion

v = stdlib_base_kernel_cos( 3.141592653589793/6.0, 0.0 );
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md [hunk 3]
```diff
@@ -96,6 +96,99 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/kernel_cos.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <math.h>
+#include <inttypes.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md:169]:
```suggestion
#include <stdlib.h>
#include <stdio.h>
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md [hunk 4]
```diff
@@ -96,6 +96,99 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+int main() {
+    double out;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/README.md:172]: Need to reorder.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+/**
+* Runs a benchmark.
+*
+* @return elapsed time in seconds
+*/
+double benchmark() {
+	double elapsed;
+	double x;
+	double y;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/benchmark/c/benchmark.c:98]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/benchmark/c/benchmark.c [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	double z;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( ( rand_double() * 2.0 ) - 1.0 ) * 0.7853981633974483;
+		y = 0.0;
+		z = stdlib_base_kernel_cos( x, y );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/benchmark/c/benchmark.c:107]:
```suggestion
		z = stdlib_base_kernel_cos( x, 0.0 );
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+#include "stdlib/math/base/special/kernel_cos.h"
+#include <stdlib.h>
+#include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/examples/c/example.c:22]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/examples/c/example.c [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+#include <stdio.h>
+
+int main() {
+    double out;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/examples/c/example.c:26]: Need to reorder.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/src/addon.c [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/kernel_cos.h"
+#include <node_api.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/src/addon.c:20]: Need to update to use binary macro.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/kernel_cos.h"
+#include "polyval_c13.c"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c:20]: Need to update to inline via script.

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+#include "polyval_c46.c"
+
+/**
+* Computes the cosine of a number on [-π/4, π/4].
+*
+* @param x    input value (in radians, assumed to be bounded by `~pi/4` in magnitude)
+* @param y    tail of `x`
+* @return	  sine value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c:28]:
```suggestion
* @return	  cosine value
```

File: lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+*
+* @example
+* double x = 1.0;
+* double y = 2.0;
+*
+* double out = stdlib_base_kernel_cos( x, y );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/kernel-cos/src/main.c:34]:
```suggestion
* double out = stdlib_base_kernel_cos( 1.0, 2.0 );
```

**Review by kgryte — APPROVED** [review]
As the requested changes are not blocking, will merge and clean-up on my end. Thanks, @Pranavchiku!

**Pull request merged**